### DISCUSSION TITLE
Hajk 3.8 Release Candidate 2

### DISCUSSION THREAD

**Discussion by Hallbergs** [hajkmap/Hajk#962] [General]
## Why so many new commits since RC1?
We decided to merge develop to master and create a new release candidate based on that. The reasoning behind this was that we had a lot of important bug-fixes as well as several nice-to-have features merged into develop since RC1 was released. You will have the opportunity to test this release during the coming weeks, and hopefully we can release 3.8 some time during January!

---

_[303 commits](https://github.com/hajkmap/Hajk/compare/v3.7.0...v3.8.0-rc.1) and changes in 201 files since the 3.7 release – here's Hajk 3.8! This release was brought to you by @Hallbergs  and @jacobwod . Please see the list of commits for all contributors._

# Changelog

## Since Release Candidate 1
* Print enhancements (#908)
* FIR plugin added (Varberg-specific implementation, #855)
* New generic KML model (various commits starting with 68bdb12c27d484b1ba2ff78ba38343d1b74da9ce), including new KML Writer from OpenLayers (#928)
* New generic Draw model (various commits starting with 1cd2facd9354828133165c41983dc22cc8917327) (#931)
* .NET-based proxy intended for rewriting requests to GeoServer ported from Hajk 2 (#930)
* Open Sans added to default build (Göteborg-specific implementation, #934)
* Correct alpha-numeric sorting of results in Search, (#947) 
* Possibility to turn off min/max zoom level warnings on a per-layer basis (#948)
* Search: show labels on features in map directly when searching (#949)
* GeoSuite plugin added (#950)
* Search: option to limit search to only visible layers (including necessary changes in Admin UI to specify search fields, more - #914)
* Docker documentation and improvements (#960)
* Additional functionality added to FME Server plugin (#955)

Bug fixes: #909, #916, #919, #920, #921, #923, 7561c271e1d253eaaa7fa9dca470d4d401042bc2, efc0aef50ea3876bd384ade587ca42b8a4cc80b5, #932, #936, #938, 4568254c658006ced0c01c5622dbceee0f61c2e6, #940, 0668b4bffe17cd5e5511fd20012c1efc417d6d50, e83b73a3e3be59a24d719e3a4f06bef241ea5225, #958,

### Added FME Server plugin. #750
This opens up new possibilities for integration with Safe's FME Server application. 
![MicrosoftTeams-image](https://user-images.githubusercontent.com/110222/141750991-01aca969-cc87-4567-8983-070dcecee5b7.png)

### Search plugin respects different response types. #882
Search plugin now respects different info formats (GML 2/3/3.2 and GeoJSON). Makes Hajk work with QGIS Server's WFS implementation. 

### Improvements in Edit. #884
More data types are allowed in the plugin.

### Improve Hajk's accessibility. #877
It's now possible to navigate the map using keyboard. 

### Make Client UI user-aware. #899
The added UI Control is just a proof-of-concept, with virtually no functionality other than displaying the user's name in the UI. However, this opens up a lot of possibilities for plugins to optionally read provided information from Active Directory. 

## New features

* Major improvements for **QGIS Server** users, including better support for `GetLegendGraphics` (#879) and `FeatureInfo` (#880) requests. 
* **TLS** Support in **Backend** makes it possible to authenticate via **LDAPS**. #893
* Allow Admin to enable Home button #876
* **Optional info** is now shown also for **background layers**. #808
* Make it possible to disable View rotation, among other **default OpenLayers interactions** #875
* **Location plugin**: make it more like other map apps #863
* Better **EDP integration** support #861
* Added endpoint **`/fmeproxy`** as a new controller https://github.com/hajkmap/Hajk/pull/900

## Bug fixes and minor features
*  #869, #864, #854, #860, #859, #894, #898, #897, #901, #902.

# Checklist for admins
- [ ] Verify that search works with different kinds of sources. **Previous to this release, search always used GeoJSON, but as of this release the setting from Admin UI is respected. Since the default setting in Admin UI has always been GML2, _that's what will be used by default from this release_. If you want to stay with GeoJSON: make sure to change the response type for your search sources.**
- [ ] Verify that FeatureInfo ("info click") works, for multiple layer types.
- [ ] Ensure that the Edit plugin works with your current configuration. The plugin has seen substantial update in this release. 
- [ ] If you use the Active Directory authentication in NodeJS backend, you now have the possibility to supply certificates and use LDAPS. Check out details in `.env`. 
- [ ] If you're planning to use the new FME Server plugin, make sure that the `url_base` key is set in Admin UI's `config.json`. (This has been added to `config.json` in this release.)
- [ ] If you'd like to use the more compact version of the Location plugin, ensure to set the `target` properly to `control` in Admin UI for that plugin. 

---

# Downloads
There are multiple ways to get the latest Hajk up and running, including Docker container.

## Pre-compiled releases
As usual, we have prepared **three different** versions of the pre-compiled solution (scroll down for download links):
1. [Hajk 3.8.0-rc.2 NodeJS edition](https://github.com/hajkmap/Hajk/releases/download/v3.8.0-rc.2/hajk-v3.8.0-rc.2-nodejs.zip)
1. [Hajk 3.8.0-rc.2 Simple edition](https://github.com/hajkmap/Hajk/releases/download/v3.8.0-rc.2/hajk-v3.8.0-rc.2-simple.zip)
1. [Hajk 3.8.0-rc.2 .NET edition](https://github.com/hajkmap/Hajk/releases/download/v3.8.0-rc.2/hajk-v3.8.0-rc.2-dotnet.zip)

## Docker container
The container is available on Docker Hub: https://hub.docker.com/r/hallbergs/hajk

# Installation
Refer to our recently updated [installation guide](https://github.com/hajkmap/Hajk/wiki/Installation-guide-%28for-pre-packaged-releases%29).

<hr /><em>This discussion was created from the release <a href='https://github.com/hajkmap/Hajk/releases/tag/v3.8.0-rc.2'>Hajk 3.8 Release Candidate 2</a>.</em>

**Comment by WCarlsson** [comment]
We found one small bug in draw-plugin. 
If you draw a point and show measurments (coordinates) and then move the point the coordinates still show the coordinates for the old location.This might be a non-issue since the sketch-plugin is coming.

Besides that we see big improvements when using it in our environment: NodeJS-backend and QGIS-server.
Awesome job!

**Reply by jacobwod** [reply]
Happy to hear that!

Regarding the Draw bug: @Hallbergs is working on a complete rewrite, called Sketch. I'm sure that this bug will be nonexistent in the upcoming plugin, so stay tuned. 😃
